src/main/java/com/aws/greengrass/lifecyclemanager/KernelLifecycle.java

```diff
@@ -123,6 +166,69 @@ public void launch() {
 
         logger.atInfo().setEventType("system-start").addKeyValue("main", kernel.getMain()).log();
         startupAllServices();
+
+    }
+
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    private void executeProvisioningPlugins(List<DeviceIdentityInterface> provisioningPlugins) {
+        logger.atDebug().log("Found provisioning plugins to run");
+        RetryUtils.RetryConfig retryConfig = RetryUtils.RetryConfig.builder()
+                .maxAttempt(MAX_PROVISIONING_PLUGIN_RETRY_ATTEMPTS)
+                .retryableExceptions(Collections.singletonList(RetryableProvisioningException.class))
+                .build();
+        ExecutorService executorService = kernel.getContext().get(ExecutorService.class);
+        executorService.execute(() -> {
```

Comment:
get a future back so we know if they complete or not?

Reply:
Where will you be checking that? If we wait on it after starting all the services then launch() can potentially get into waiting indefinitely. And if we put timeout then it will fail after a hard limit, which is not desirable. We probably need a configurable timeout for provisioning plugin?